packages/gestalt/src/SideNavigation.css

```diff
@@ -14,13 +15,39 @@
 
 .fullHeight {
   height: 100%;
-  overflow: scroll;
+  overflow-y: scroll;
+}
+
+.contentWidthTransition {
+  transition: width 0.2s;
 }
 
 .section {
+  margin-bottom: var(--space-600);
   margin-top: var(--space-600);
 }
 
-.section:nth-child(1) {
+.section:first-child {
   margin-top: 0;
 }
+
+.section:last-child {
+  margin-bottom: 0;
+}
+
+.sectionStartDivider {
+  display: none;
+  margin-bottom: var(--space-600);
+}
+
+.sectionEndDivider {
+  margin-top: var(--space-600);
+}
+
+.liItem:not(.section) + .section .sectionStartDivider {
+  display: block;
```

Comment:
`.liItem:not(.section)` represents `TopItem` or `Group`.

For example, `TopItem` is the previous *adjacent* sibling of `Section`, the section's top divider should appear:

```jsx
<TopItem />
<Section /> // top divider appears
```

Reply:
nit Some of these comments could go in the code itself for future immediate reference. 

reading 
`.liItem:not(.section) + .section .sectionStartDivider {` isn't the most obvious so a comment would be nice

Reply:
done